lib/active_fedora/rdf/indexing_service.rb

```diff
@@ -0,0 +1,85 @@
+module ActiveFedora::RDF
+  class IndexingService
+    include Solrizer::Common
+    attr_reader :object, :opts
+    # Object must respond to
+    #   create_date
+    #   modified_date
+    #   has_model
+    #   id
+    #   to_json
+    #   attached_files
+    #   []
+    # and it's class must respond to
+    #   inspect
+    #   outgoing_reflections
+    # @param obj [ActiveFedora::Base] the object to build an solr document for
+    # @param opts [Hash]
+    # @option opts [String] :name the prefix for the solr keys
+    def initialize(obj, opts = {})
+      @object = obj
+      @opts = opts
+    end
+
+    # Creates a solr document hash for the rdf assertions of the {#object}
+    # @yield [Hash] yields the solr document
+    # @return [Hash] the solr document
+    def generate_solr_document
+      solr_doc = add_assertions
+      yield(solr_doc) if block_given?
+      solr_doc
+    end
+
+    protected
+
+      def add_assertions(solr_doc = {})
+        fields.each do |field_key, field_info|
+          values = resource.get_values(field_key)
+          Array(values).each do |val|
+            append_to_solr_doc(solr_doc, field_key, field_info, val)
+          end
+        end
+        solr_doc
+      end
+
+      # Override this in order to allow one field to be expanded into more than one:
+      #   example:
+      #     def serialize_to_solr(solr_doc, field_key, field_info, val)
```

Comment:
This comment seems confusing - the method names don't match. Should we be advocating overriding a protected method?